src/utils/database.ts

```diff
@@ -0,0 +1,40 @@
+import { existsSync, readFileSync, writeFileSync, unlinkSync } from 'fs';
+
+export class Database {
+  defaultFilePath: string;
+
+  constructor() {
+    this.defaultFilePath = 'src/utils/data.json';
+  }
+
+  async hasGameFile(filePathOverride?: string): Promise<boolean> {
+    return existsSync(filePathOverride || this.defaultFilePath);
+  }
+
+  async load(filePathOverride?: string): Promise<object> {
+    try {
+      const data = await readFileSync(
+        filePathOverride || this.defaultFilePath,
+        'utf-8'
+      );
```

Comment:
fs.*Sync functions do not return promises and are blocking on purpose. We decided to use them so that the api wasn't async. 

Theses errors should have come up automatically. I created a [PR](https://github.com/normal-devs/klicker-knight/pull/6) to enforce them. They should help clean up this PR as it looks like it's returning a promise but calling `.then` on the result would throw an error.

Reply:
Ah. Yeah, those new lint rules helped.